components/ContentTemplates/ButtonsTemplate.tsx

```diff
@@ -1,5 +1,428 @@
+				</p>
+				<ul>
+					<li>
+						Product 1{" "}
+						<button className="exampleButton" onClick={handleAddToBasketButton}>
+							Add to basket
+						</button>
+					</li>
+					<li>
+						Product 2{" "}
+						<button className="exampleButton" onClick={handleAddToBasketButton}>
+							Add to basket
+						</button>
+					</li>
+					<li>
+						Product 3{" "}
+						<button className="exampleButton" onClick={handleAddToBasketButton}>
+							Add to basket
+						</button>
+					</li>
+					<li>
+						Product 4{" "}
+						<button className="exampleButton" onClick={handleAddToBasketButton}>
+							Add to basket
+						</button>
+					</li>
+					<li>
+						Product 5{" "}
+						<button className="exampleButton" onClick={handleAddToBasketButton}>
+							Add to basket
+						</button>
+					</li>
+				</ul>
+				<p>We could use CSS to visibly hide some extra text:</p>
+				<CodeBlock
+					codeSnippet={`.visibly-hidden {
+	clip: rect(0 0 0 0); 
+	clip-path: inset(50%);
+	height: 1px;
+	overflow: hidden;
+	position: absolute;
+	white-space: nowrap; 
+	width: 1px;
+}
+					
+<button type="button" onclick="handleClick()">Add to basket
+	<span class="visibly-hidden">Product 1</span>
+</button>`}
+				/>
+				<p>
+					This would then read out "Add to basket product 1" to screen reader
+					users.
+				</p>
+				<p>
+					However, it's important to be careful of word order when using this
+					technique. If we had instead inserted the product name in the middle
+					of the sentence, for example: "Add product 1 to basket" this can cause
+					problems for users of speech input software.
+				</p>
+				<p>We could also use an aria-label.</p>
+				<CodeBlock
+					codeSnippet={`<button type="button" onclick="handleClick()" aria-label="Add to basket product 1">Add to basket</button>`}
+				/>
+				<p>
+					Again, it is important that the start of the aria-label matches with
+					the visible text on the button to help users of speech input software
+					be able to activate the button.{" "}
+				</p>
+			</TemplateSection>
+			<TemplateSection
+				sectionName="textAndIconButtons"
+				title="Text and Icon Buttons">
+				<p>
+					Some buttons have both text and icons inside. The icon can help
+					complement the text and aid understanding. To prevent the accessible
+					name from repetition, eg "Button, save icon, save", the icon should be
+					given an empty(null) alt attribute or the state aria-hidden="true"
+				</p>
+				<button
+					className="exampleButton"
+					type="button"
+					onClick={handleTextAndIconButton}>
+					<FaSave color="black" size="1rem" aria-hidden="true" /> Save
+				</button>
+				<CodeBlock
+					codeSnippet={`<button type="button" onclick="handleClick()">
+	<img src="icon-url.png" alt="" />
+		Text
+</button>`}
+				/>
+				<CodeBlock
+					codeSnippet={`<button type="button" onclick="handleClick()">
+	<img src="icon-url.png" aria-hidden="true" />
+		Text
```

Comment:
Just a curiosity, I see `Text` is indented (same as above). Since it's not inside an open/closed tag, did you indent it for readability?

Reply:
I did. I think this code block component needs a bit of work.